processing/src/main/java/io/druid/query/BySegmentQueryRunner.java

```diff
@@ -61,7 +62,11 @@ public BySegmentQueryRunner(
                   new BySegmentResultValueClass<T>(
                       results,
                       segmentIdentifier,
-                      query.getIntervals().get(0)
+                      Iterables.getOnlyElement(
+                          Iterables.getOnlyElement(query.getDataSources())
```

Comment:
Query should still have a method `getDataSource()`, which fails if there are multiple data sources.